sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/ConfigProperties.java

```diff
@@ -93,12 +160,46 @@ public interface ConfigProperties {
   List<String> getList(String name);
 
   /**
-   * Returns a map-valued configuration property. The format of the original value must be
-   * comma-separated for each key, with an '=' separating the key and value. For instance, <code>
+   * Returns a List value configuration property.
+   *
+   * @see ConfigProperties#getList(String name)
+   * @return a List configuration property or {@code defaultValue} if a property with {@code name}
+   *     has not been configured.
+   * @throws ConfigurationException if the property is not a valid string.
+   */
+  default List<String> getList(String name, List<String> defaultValue) {
+    List<String> value = getList(name);
+    return (value == null || value.isEmpty()) ? defaultValue : value;
+  }
+
+  /**
+   * Returns a Map configuration property. The format of the original value must be comma-separated
+   * for each key, with an '=' separating the key and value. For instance, <code>
    * service.name=Greatest Service,host.name=localhost</code> Empty values will be removed.
    *
    * @return an empty map if the property has not been configured.
    * @throws ConfigurationException for malformed map strings.
    */
   Map<String, String> getMap(String name);
+
+  /**
+   * Returns a Map value configuration property.
+   *
+   * @see ConfigProperties#getMap(String name)
+   * @return a Double-valued configuration property or {@code defaultValue} if a property with
+   *     {@code name} has not been configured.
+   * @throws ConfigurationException if the property is not a valid string.
+   */
+  default Map<String, String> getMap(String name, Map<String, String> defaultValue) {
+    Map<String, String> value = getMap(name);
+    return (value == null || value.isEmpty()) ? defaultValue : value;
+  }
+
+  /**
+   * Returns defaultValue if value is null, otherwise value. This is an internal method that should
+   * not be broadly used.
+   */
+  default <T> T defaultIfNull(@Nullable T value, T defaultValue) {
```

Comment:
Let's move this to a package-private class and then wouldn't need the scary ignorable javadoc. Or consider just inlining - it's ok, but slightly harder to reason about the behavior of this method with the primitive types / autoboxing.